chime/view_functions.py

```diff
@@ -435,10 +439,23 @@ def decorated_function(*args, **kwargs):
             return redirect(redirect_url)
 
         auth_data_href = current_app.config['AUTH_DATA_HREF']
-        if not is_allowed_email(get_auth_data_file(auth_data_href), email):
+        last_check = session.get('auth_check', {}).get('last_check', 0.0)
+        last_result = session.get('auth_check', {}).get('last_result')
+
+        if last_result is True and (last_check + AUTH_CHECK_LIFESPAN) > time.time():
+            Logger.info("Email still allowed.")
+
+        elif not is_allowed_email(get_auth_data_file(auth_data_href), email):
+            Logger.debug('Remembering that email was not allowed')
+            session['auth_check'] = dict(last_check=time.time(), last_result=False)
```

Comment:
Do we need to cache the negative case? It seems to me that people won't be doing this often, and if they are, I'm ok that it's slow, as presumably they're doing it because the administrator changed something and we want fast response.